### PR TITLE
Add GET route to homography, adjust argument finding for GET vs POST

### PR DESCRIPTION
**Pull request by dcelik**
This PR adds the following functionality:
* baseHandler now contains definitions for MB and GB to eliminate redefinition in classes that need it
* baseHandler now has a method find_argument(self, arg_name) that will search for the argument in either the header or the body depending on which method type (POST, GET, etc. ) is used. NOTE: This method cannot be used before the request headers have been loaded. As such it cannot be used in initialize but can be used in prepare or later. It also should not be used for any handlers with the stream_request_body decorator, as it will not generate a body and as such will cause errors when trying to search a POST request for a body argument
* EDIT: configHomography now supports a GET route that will writes the contents of the homography.txt file of the project as a JSON in the response. 

This PR modifies the following functionality:
* Any function call that could be called in multiple HTTP methods now uses find_arguments instead of get_body_arguments. For functions with no GET route, it will return the same behavior as before. For the case in which get is sent to a method without a GET route, the handler should return an error instead. 
* the write_file_stream method in baseHandler now has more robust error checking for cases in which a file has not been created. The try-catch could also be expanded instead for similar results.
* EDIT: configHomography now checks for any errors in converting the points from strings to a list, and updates the status appropriately. It also does checks to make sure that the points obey the following constraints: List of lists, where each inner list contains 2 floats [x_coord, y_coord]; both lists are the same length; the lists contain at least 4 points.

* testConfig was leaking project structure information on return, as file_name is the path to our file, rather than just the name of it, as desired. A consistent manner of checking method types was also applied: self.request.method.lower() will ensure that we don't get capitalization errors anywhere.
* uploadVideo now more clearly indicates the max_body_size by using the baseHandler stored definition of GB.

The corresponding GUI code can be found in [this PR](https://github.com/santosfamilyfoundation/SantosGUI/pull/74).

### CODE REVIEW

File: app/handlers/baseHandler.py [hunk 1]
```diff
@@ -1,10 +1,26 @@
...
 
 class BaseHandler(tornado.web.RequestHandler):
     def initialize(self):
         self.error_message = None
+        self.MB = 1024*1024
+        self.GB = 1024*self.MB
+
+    def find_argument(self, arg_name):
+        method_type = self.request.method.lower()
+        if method_type == 'post':
```
Comment (runnersaw) [inline app/handlers/baseHandler.py]: We should document this in apidocs

File: app/handlers/configHomography.py [hunk 2]
```diff
@@ -39,15 +43,42 @@ def post(self):
...
-        #TODO: Put literal_eval here and use try/catch
-        aerial_pts = self.get_body_argument('aerial_pts')
-        camera_pts = self.get_body_argument('camera_pts')
-
-        homography, mask = cv2.findHomography(\
-                            np.array(literal_eval(camera_pts)),\
-                            self.up_ratio*np.array(literal_eval(aerial_pts)))
-        np.savetxt(os.path.join(project_dir,'homography','homography.txt'),homography)
+        aerial_pts = literal_eval(self.get_body_argument('aerial_pts'))
+        camera_pts = literal_eval(self.get_body_argument('camera_pts'))
```
Comment (runnersaw) [inline app/handlers/configHomography.py]: I feel like better error checking should be done here

File: app/handlers/retrieveResults.py [hunk 3]
```diff
@@ -24,7 +24,7 @@ class RetrieveResultsHandler(BaseHandler):
     """
 
     def get(self):
-        identifier = self.get_body_argument('identifier')
+        identifier = self.get_argument('identifier')
```
Comment (runnersaw) [inline app/handlers/retrieveResults.py]: `find_argument`

File: app/handlers/status.py [hunk 4]
```diff
@@ -23,8 +23,8 @@ class StatusHandler(BaseHandler):
 
     @apiError error_message The error message to display. (Will return unique error message if object tracking has NOT been run on specified project)
     """
-    def post(self):
-        identifier = self.get_body_argument('identifier')
+    def get(self):
+        identifier = self.get_argument('identifier')
```
Comment (runnersaw) [inline app/handlers/status.py]: `find_argument`

**Review by runnersaw — CHANGES_REQUESTED** [review]

File: app/handlers/configHomography.py [hunk 5]
```diff
@@ -39,15 +43,42 @@ def post(self):
...
         self.write_homography_files()
         StatusHelper.set_status(self.identifier, Status.Type.CONFIG_HOMOGRAPHY, Status.Flag.COMPLETE)
-        self.finish("Upload Homography")
+        self.finish()
+
+    def get(self):
+        self.set_header('Content-Type', 'application/octet-stream')
+        self.set_header('Content-Description', 'File Transfer')
+        self.set_header('Content-Disposition', 'attachment; filename=homography.txt')
+        self.write_file_stream(\
```
Comment (runnersaw) [inline app/handlers/configHomography.py]: I think we should read this and return JSON instead

File: app/handlers/status.py (hunk 4 above)
Comment (dcelik) [reply app/handlers/status.py]: If we are in a get(Self) or post(self) we can use get_argument or get_body_argument safely. If we want to make everything find_argument for consistency, that is fine  (there is no change in behavior).

File: app/handlers/baseHandler.py (hunk 1 above)
Comment (dcelik) [reply app/handlers/baseHandler.py]: The apidocs has been updated to reflect the changes in GET and POST. According to the apidocs website (credit @runnersaw) developers should be able to understand that GET takes parameters in the url and POST takes parameters in the body.

**Review by runnersaw — APPROVED** [review]
:ship: :it:

**Review by jabb1123 — APPROVED** [review]

**Pull request merged**